### PR TITLE
fix(lambda-nodejs): pnpm exec command

### LINKED ISSUE #14757: (aws-lambda-nodejs): support pnpm
`aws-lambda-nodejs` currently only supports npm and yarn, but it would be nice to also support pnpm.

E.g. see https://github.com/aws/aws-cdk/blob/master/packages/%40aws-cdk/aws-lambda-nodejs/lib/function.ts#L55.

### Use Case

pnpm is a powerful and fast alternative dependency manager in the Node ecosystem - with a growing popularity.

### Proposed Solution

E.g. `depsLockFilePath` should also support `pnpm-lock.yaml`.

If you don't want to support pnpm it would be nice to get some hint about this as soon as possible (- currently it fails during deployment).

(Stupid question... I just looked at the source code and it looks like `depsLockFilePath` isn't really used for something meaningful - at least nothing lock file specific, besides a small yarn/npm difference check. It's more used to get the project path. Maybe two dedicated settings `projectPath` and `packageManager` would be more useful?)

### Other

* [ ] :wave: I may be able to implement this feature request
* [ ] :warning: This feature might incur a breaking change

---

This is a :rocket: Feature Request

### PR DESCRIPTION
**Pull request by relm923**
Fix #14757

Expands on #14772

Currently get the following error when bundling with `pnpm-lock.yaml`
```
ERROR   ERROR  Unknown options: 'bundle', 'target', 'platform', 'outfile', 'external:aws-sdk'
```

----

Switch from `pnpm run esbuild` to `pnpm exec esbuild --` to properly enable esbuild with `pnpm`

`pnpm run` only supports running commands defined in the package's manifest file - [docs](https://pnpm.io/cli/run)

`pnpm exec` supports executing a shell command in scope of a project - [docs](https://pnpm.io/cli/exec)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-lambda-nodejs/lib/package-manager.ts [hunk 1]
```diff
@@ -60,6 +60,7 @@ export class PackageManager {
       os.platform() === 'win32' ? `${runCommand}.cmd` : runCommand,
       ...runArgs,
       bin,
+      ...(this.lockFile === PackageManager.PNPM.lockFile ? ['--'] : []),
```
Comment (relm923) [inline packages/@aws-cdk/aws-lambda-nodejs/lib/package-manager.ts:63]: Required to pass parameters to bin command

File: packages/@aws-cdk/aws-lambda-nodejs/lib/package-manager.ts [hunk 2]
```diff
@@ -26,7 +26,7 @@ export class PackageManager {
   public static PNPM = new PackageManager({
     lockFile: 'pnpm-lock.yaml',
     installCommand: ['pnpm', 'install'],
-    runCommand: ['pnpm', 'run'],
+    runCommand: ['pnpm', 'exec'],
```
Comment (jogold) [inline packages/@aws-cdk/aws-lambda-nodejs/lib/package-manager.ts:29]: Getting `The "pnpm exec" command currently only works with the "-r" option` with only `exec`.

File: packages/@aws-cdk/aws-lambda-nodejs/lib/package-manager.ts (hunk 1 above)
Comment (jogold) [reply packages/@aws-cdk/aws-lambda-nodejs/lib/package-manager.ts:63]: Let's introduce `argsSeparator?: string` in `PackageManager`.

**Review by jogold — COMMENTED** [review]
Good catch @relm923! Thanks.

See my comments.

File: packages/@aws-cdk/aws-lambda-nodejs/lib/package-manager.ts (hunk 2 above)
Comment (relm923) [reply packages/@aws-cdk/aws-lambda-nodejs/lib/package-manager.ts:29]: Can you try:
```sh
pnpm --version

cd aws-cdk/packages/@aws-cdk/aws-lambda-nodejs

pnpm exec esbuild -- --bundle test/integ-handlers/ts-handler.ts
```

Comment (jogold) [reply packages/@aws-cdk/aws-lambda-nodejs/lib/package-manager.ts:30]: OK it's working. For some reason not in the Docker image...

**Review by jogold — APPROVED** [review]
@eladb LGTM

**Review by eladb — APPROVED** [review]

**Pull request merged**